### PR TITLE
Add guest post on TS

### PR DESCRIPTION
**Pull request by NotWoods**
Let me know what needs to be tweaked here. I added a bit at the top based on similar guest post text [in CSS Tricks](https://css-tricks.com/templating-languages-and-wordpress/).

Once this is merged a hacktoberfest-accepted label would be a nice to have :)

### CODE REVIEW

**Comment by evantahler** [comment]
That was ridiculously fast! 

The post looks great! I want to do https://github.com/grouparoo/www.grouparoo.com/issues/18 first, to add a little more author information.  Thanks for providing your photo... would you mind writing a short description for yourself in a comment on https://github.com/grouparoo/www.grouparoo.com/issues/18.

This way, the "The following is a guest post..." section shouldn't be needed, as you'll have more rich "Author" area on the page.

**Comment by evantahler** [comment]
Also, I'm glad you found this site easy to work with! If you've got any suggestions about cloning & running the project locally, please let me know!

**Comment by evantahler** [comment]
ok! https://github.com/grouparoo/www.grouparoo.com/pull/20 is merged in and now there are `<AuthorBox />` components on the pages.  If you rebase this PR against the `master` branch, you can add yourself to [`data/authors.ts`](https://github.com/grouparoo/www.grouparoo.com/blob/master/data/authors.ts) so your'll have a richer entry about yourself on the post & author page.  

Please add your image to `/public/images/authors` rather than `/public/images/team`

Thank you!

**Comment by NotWoods** [comment]
Moved stuff around! The repo was very easy to get started with, the info in the README made it easy to get going

File: components/blog/authorBox.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ export default function AuthorBox({
           <Image
             style={{ width: "100%", maxWidth: 200 }}
             roundedCircle
-            src={require(`../../public/images/team/${author.slug}.png`)}
+            src={require(`../../public/images/authors/${author.slug}.png`)}
           />
```
Comment (evantahler) [inline components/blog/authorBox.tsx:21]: Thanks!

**Review by evantahler — APPROVED** [review]

File: pages/blog/any-wreaks-havoc-on-overloads.mdx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+let obj: any = { x: 0 };
+// None of these lines of code are errors
+const foo: any = obj.foo();
+obj();
+obj.bar = 100;
+```
+
+## What are function overloads?
+
+TypeScript has another [neat feature called function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads). Some JavaScript functions return different results based on the arguments you supply, and this can be represented in TypeScript by written multiple function types on top of each other. Only one function signature can match at a time. The matching overload is determined by the arguments you supply to he function. [The first applicable overload will always be chosen](https://github.com/microsoft/TypeScript/issues/1860#issuecomment-72154737).
```
Comment (andyjih) [inline pages/blog/any-wreaks-havoc-on-overloads.mdx]: "to he" -> "to the"

**Review by andyjih — CHANGES_REQUESTED** [review]
Looks great! Just a small typo

**Review by andyjih — APPROVED** [review]

**Comment by andyjih** [comment]
👍

**Pull request merged**